### PR TITLE
Client side testing boilerplate

### PR DESCRIPTION
**Pull request by justinvdm**
Since the new Go UI will be needing considerably more complex client side logic than the previous UI, we will need to include client side testing.

### CODE REVIEW

**Comment by smn** [comment]
+100

**Comment by justinvdm** [comment]
I'm not sure how, but we will probably need to include these client side tests in the CI we are now doing with Travis. I'll look into doing that in the PR associated to this issue.

File: go/base/static/js/test/runner.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <script src="../vendor/jquery.js"></script>
+    <script src="../vendor/jquery.ui.js"></script>
+    <script src="../vendor/jquery.jsPlumb.js"></script>
+    <script src="../vendor/underscore.js"></script>
+    <script src="../vendor/backbone.js"></script>
+
+    <script src="../src/init.js"></script>
+    <script src="../src/campaigns/init.js"></script>
+    <script src="../src/campaigns/setup/init.js"></script>
+    <script src="../src/campaigns/setup/interactive.js"></script>
```
Comment (justinvdm) [inline go/base/static/js/test/runner.html:27]: We can avoid having to specify each and every module, vendor lib and test module if we concat our modules using something like [grunt](http://gruntjs.com/). I know that the current plan is to use django-pipeline, which appears to concatenate the files at runtime. This means we will need to continue specifying everything here. I'm fine with that, though. Any thoughts or suggestions?

**Comment by justinvdm** [comment]
One concern of mine is that the tests are in the static dir (`go/base/static/js/test`), so they are publically viewable, which is a bit wierd. I could move them out of static, but then they are far away from the code they are testing, which isn't good. Any thoughts or suggestions?

**Comment by smn** [comment]
The whole repository is open so it's always publicly viewable, not sure what the problem is with that?

**Comment by justinvdm** [comment]
Not really worried about security at all. I'm more worried that its bad practice, but I'm probably overthinking things.

**Comment by smn** [comment]
I'm happy with them living in `static` for the time being until we find a better way of doing it or find it isn't a problem after all.

**Comment by peterp** [comment]
Pipelines creates a unique filename for the the minified and combined library if changes are detected. (In order to overcome agressive caching.)

I'm guessing that this might be a problem as you'll have to modify the tests if you want to test against the files in production.

**Comment by peterp** [comment]
:+1:

**Pull request merged**